content/guides/component-testing/framework-configuration.md

```diff
@@ -0,0 +1,328 @@
+
+In this example we specify the `componentFolder` as `components`, the default for Nuxt.
+
+```json
+{
+  "testFiles": "**/*.spec.js",
+  "componentFolder": "components"
+}
+```
+
+Finally, add a component and test:
+
+```html
+<!-- components/mountains.vue -->
+
+<template>
+  <p v-if="$fetchState.pending">Fetching mountains...</p>
+  <p v-else-if="$fetchState.error">An error occurred :(</p>
+  <div v-else>
+    <h1>Nuxt Mountains</h1>
+    <ul>
+      <li v-for="mountain of mountains">{{ mountain.title }}</li>
+    </ul>
+  </div>
+</template>
+
+<script>
+  export default {
+    data() {
+      return {
+        mountains: [],
+      }
+    },
+    async fetch() {
+      this.mountains = await fetch(
+        'https://api.nuxtjs.dev/mountains'
+      ).then((res) => res.json())
+    },
+  }
+</script>
+```
+
+```js
+// components/mountains.spec.js
+import { mount } from '@cypress/vue'
+import Mountains from './mountains.vue'
+
+describe('Mountains', () => {
+  it('shows a load state', () => {
+    mount(Mountains, {
+      mocks: {
+        $fetchState: {
+          pending: true,
+        },
+      },
+    })
+
+    cy.get('p').contains('Fetching mountains...')
+  })
+
+  it('shows a failed state', () => {
+    mount(Mountains, {
+      mocks: {
+        $fetchState: {
+          error: true,
+        },
+      },
+    })
+
+    cy.get('p').contains('An error occurred :(')
+  })
+
+  it('shows a failed state', () => {
+    mount(Mountains, {
+      data() {
+        return {
+          mountains: [{ title: 'Mt Everest' }],
+        }
+      },
+      mocks: {
+        $fetchState: {},
+      },
+    })
+
+    cy.get('li').contains('Mt Everest')
+  })
+})
+```
+
+Because Cypress mounts components in islation, Nuxt specific APIs are generally _not_ applied. In this example, the `fetch` hook is not automatically applied, so we used the `mocks` mounting option to specify the three component states (loading, error and success) and test each one in isolation.
+
+Start Cypress with `npx cypress open-ct` - the test runner will open. Select your test to execute it and see the rendered output. You can also run the tests without opening a browser with `npx cypress run-ct`.
+
+## Vite Based Projects (Vue, React)
+
+Cypress also ships a Vite based dev server, as opposed to a Webpack based on like the other examples on this page. This example uses a Vite project with React, created via `npm init @vitejs/app my-react-app -- --template react`. The configuration instructions are the same for Vue.
+
+Inside of `cypress/plugins/index.js`, configure Cypress to use the Vite dev server:
+
+```js
+const path = require('path')
+const { startDevServer } = require('@cypress/vite-dev-server')
+
+module.exports = (on, config) => {
+  on('dev-server:start', (options) => {
+    return startDevServer({
+      options,
+      viteConfig: {
+        configFile: path.resolve(__dirname, '..', '..', 'vite.config.js'),
```

Comment:
I've using it yesterday and I add type error on the configFile which is UserConfig and not InlineConfig (which permits configFile attribute)

Reply:
Do you mean an inline vite config?